lightning/src/ln/outbound_payment.rs

```diff
@@ -1771,6 +1787,75 @@ impl OutboundPayments {
 	}
 }
 
+fn set_max_path_length(
+	route_params: &mut RouteParameters, recipient_onion: &RecipientOnionFields,
+	keysend_preimage: Option<PaymentPreimage>, best_block_height: u32,
+) -> Result<(), ()> {
+	const PAYLOAD_HMAC_LEN: usize = 32;
+	const UNBLINDED_INTERMED_PAYLOAD_LEN: usize = PAYLOAD_HMAC_LEN
+		+ 1 + 1 + 8 // amt_to_forward
+		+ 1 + 1 + 4 // outgoing_cltv
+		+ 1 + 1 + 8; // short_channel_id
+
+	let num_reserved_bytes = match &route_params.payment_params.payee {
+		Payee::Blinded { route_hints, .. } => {
+			let largest_path = route_hints
```

Comment:
Note: currently I’m evaluating the max path length based on the largest blinded path, i.e. the blinded path that limits us the most. Therefore, if some blinded paths are larger than others, this could unnecessarily limit us in pathfinding. We could set the max path length on a per-blinded-path basis? In practice, ISTM that blinded paths within the same invoice are likely to be of similar sizes to each other.